common/configParser_test.go

```diff
@@ -46,46 +316,179 @@ func TestLoadGasScheduleConfig(t *testing.T) {
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString(testString)
+		assert.Nil(t, err)
+
+		roundActivationConfig, err := common.LoadRoundConfig(filePath)
+		assert.Nil(t, err)
+
+		if _, err = os.Stat("testEnableRounds.toml"); err == nil {
+			_ = os.Remove("testEnableRounds.toml")
+		}
+
+		require.Equal(t, &config.RoundConfig{}, roundActivationConfig)
+	})
+}
+
+func TestLoadEpochConfig(t *testing.T) {
+	t.Parallel()
+
+	t.Run("invalid file should error", func(t *testing.T) {
+		t.Parallel()
+
+		conf, err := common.LoadEpochConfig("testFile01")
+		assert.Nil(t, conf)
+		assert.Error(t, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		testString := `
+[EnableEpochs]
+    SCDeployEnableEpoch = 1
+`
+
+		filePath := path.Join(t.TempDir(), "testFile02")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString(testString)
+		assert.Nil(t, err)
+
+		conf, err := common.LoadEpochConfig(filePath)
+		assert.NotNil(t, conf)
+		assert.Nil(t, err)
+		require.Equal(t, uint32(1), conf.EnableEpochs.SCDeployEnableEpoch)
+	})
+}
+
+func TestGetSkBytesFromP2pKey(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty file name should early return", func(t *testing.T) {
+		t.Parallel()
+
+		key, err := common.GetSkBytesFromP2pKey("")
+		assert.Equal(t, []byte{}, key)
+		assert.NoError(t, err)
+	})
+	t.Run("invalid file should return nil", func(t *testing.T) {
+		t.Parallel()
+
+		key, err := common.GetSkBytesFromP2pKey("testFile01")
+		assert.Equal(t, []byte{}, key)
+		assert.NoError(t, err)
+	})
+	t.Run("invalid file content should error", func(t *testing.T) {
+		t.Parallel()
+
+		filePath := path.Join(t.TempDir(), "testFile01")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString("invalid pem file")
+		assert.Nil(t, err)
+
+		key, err := common.GetSkBytesFromP2pKey(filePath)
+		assert.Nil(t, key)
+		assert.Error(t, err)
+	})
+	t.Run("invalid secret key should error", func(t *testing.T) {
+		t.Parallel()
+
+		invalidPem := `
+-----BEGIN PRIVATE KEY for erd1ecdwux5tvanwryhr7cn5l9kc07ayquec7h2jc608kz0ycychzexsj6qw4j-----
+INVALIDiZTI4ODY1ZGQ1MzZmOWYyOGEwOTZhNmY5NmQ2MzZjMjhmNzMzNDAwNDBj
+MTU0ZDA4Nzg3MTBhOTE5ZWNlMWFlZTFhOGI2NzY2ZTE5MmUzZjYyNzRmOTZkODdm
+YmE0MDczMzhmNWQ1MmM2OWU3YjA5ZTRjMTMxNzE2NGQ=
+-----END PRIVATE KEY for erd1ecdwux5tvanwryhr7cn5l9kc07ayquec7h2jc608kz0ycychzexsj6qw4j-----
+`
+		filePath := path.Join(t.TempDir(), "testFile01")
+		file, err := os.Create(filePath)
+		assert.Nil(t, err)
+
+		_, err = file.WriteString(invalidPem)
+		assert.Nil(t, err)
+
+		key, err := common.GetSkBytesFromP2pKey(filePath)
+		assert.Nil(t, key)
+		assert.Error(t, err)
+		assert.True(t, strings.Contains(err.Error(), "for encoded secret key"))
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		invalidPem := `
+-----BEGIN PRIVATE KEY for erd1ecdwux5tvanwryhr7cn5l9kc07ayquec7h2jc608kz0ycychzexsj6qw4j-----
```

Comment:
Is this a known private key (e.g. Alice)? Also, this should be `validPem`, right?

Reply:
I created it for tests only.. renamed to validPem